lib/mutant/ast/regexp/transformer/alternative.rb

```diff
@@ -0,0 +1,27 @@
+module Mutant
+  module AST
+    module Regexp
+      class Transformer
+        class Alternative < self
+          ALTERNATIVE = IceNine.deep_freeze(::Regexp::Expression::Alternative.new(0, 0, 0))
+
+          register :regexp_sequence_expression
+
+          class ExpressionToAST < Transformer::ExpressionToAST
+            def call
+              ast(*children)
+            end
+          end
+
+          class ASTToExpression < Transformer::ASTToExpression
+            def call
+              root = ALTERNATIVE.dup
```

Comment:
I'd also prefer `Object#tab` here. Mostly because we remove the "redundant" re appearance of the `root` token.

```
ALTERNATIVE.dup.tap do |root|
  root.expressions = subexpressions
end
```

Is easier to digest (for me).